### PR TITLE
rgw/sfs: rename simplefile to sfs

### PR DESCRIPTION
**Pull request by jecluis**
Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: src/rgw/rgw_sal_sfs.cc [hunk 1]
```diff
@@ -391,13 +403,13 @@ SimpleFileStore::SimpleFileStore(
                   << dendl;
```
Comment (0xavi0) [inline src/rgw/rgw_sal_sfs.cc:403]: So far, this is the only place I still found `simplefile` references.

File: src/rgw/store/sfs/bucket.h [hunk 2]
```diff
@@ -21,18 +21,18 @@
...
+class SFSObject;
 
-class SimpleFileBucket : public Bucket {
+class SFSBucket : public Bucket {
  private:
-  SimpleFileStore *store;
+  SFStore *store;
   BucketMgrRef mgr;
   const std::filesystem::path path;
   RGWAccessControlPolicy acls;
```
Comment (0xavi0) [inline src/rgw/store/sfs/bucket.h:38]: Also, here, but it's commented. I would maybe remove this line or change it if we want to keep it, so there are no traces of old `simplefile` naming

**Review by 0xavi0 — CHANGES_REQUESTED** [review]
I get a compilation issue:
```bash
/home/xavi/dev/joao/ceph/src/rgw/rgw_main.cc:635:12: error: expected primary-expression before ‘)’ token
  635 |   if (rados) { /* Supported for only RadosStore */
      |            ^
```

**Comment by jecluis** [comment]
weird, I didn't hit any compilation issues. oh well, I'll check it out. :shrug: 

thanks! :)

**Comment by 0xavi0** [comment]
In this line I would remove the comment or change it to be SFS of we want to keep it. 

https://github.com/jecluis/ceph/blob/96ebf991dd3f03d3348d3119fd2cc32d9b27706d/src/rgw/store/sfs/bucket.h#L39

File: src/rgw/store/sfs/sqlite/sqlite_users.cc [hunk 3]
```diff
@@ -70,8 +83,8 @@ std::vector<DBOPUserInfo> SQLiteUsers::getUsersBy(Args... args) const {
 }
 
 std::string SQLiteUsers::getDBPath() const {
-  auto rgw_simplefile_path = ceph_context_->_conf.get_val<std::string>("rgw_simplefile_data_path");
-  auto db_path = std::filesystem::path(rgw_simplefile_path) / std::string(USERS_DB_NAME);
+  auto rgw_SFS_path = ceph_context_->_conf.get_val<std::string>("rgw_SFS_data_path");
```
Comment (0xavi0) [inline src/rgw/store/sfs/sqlite/sqlite_users.cc:86]: this one should be `rgw_sfs_data_path`, otherwise it crashes when accessing the DB path

**Review by 0xavi0 — CHANGES_REQUESTED** [review]

**Review by 0xavi0 — APPROVED** [review]
After fixing the 2 major issues it passes the smoke tests.
I could not find any other `simplefile` reference, so LGTM.

**Comment by giubacc** [comment]
> @giubacc try adding the `rgw_sfs_data_path` parameter. Maybe there is something wrong with the default value for it... I will also investigate.
> 
> for example:
> 
> ```shell
> --rgw_sfs_data_path=/tmp/sfs
> ```

That's weird, I have try every combination `-` and `--` but that parameter `rgw_sfs_data_path` seems having no effect on my running process.
It tries to always create the sqlile file at: `/var/lib/ceph/radosgw/`

```
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw dbstore: DB initialization full db_path("/var/lib/ceph/radosgw/dbstore-default_ns")
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw DBStore backend: Cant open /var/lib/ceph/radosgw/dbstore-default_ns.db; Errmsg - unable to open database file
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw DBStore backend: sqlite exec failed for schema(PRAGMA foreign_keys=ON); Errmsg - 
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw DBStore backend: sqlite exec failed for schema(CREATE TABLE IF NOT 
...
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw DBStore backend: CreateUserTable failed
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw DBStore backend: sqlite exec failed for schema(DROP TABLE IF EXISTS '/var/lib/ceph/radosgw/dbstore-default_ns_user_table'); Errmsg - 
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw DBStore backend: DeleteUserTable failed 
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw DBStore backend: Creation of tables failed
2022-07-04T15:12:12.889+0000 7ffff3fcce00  0 rgw DBStore backend: DB successfully initialized - name:/var/lib/ceph/radosgw/dbstore-default_ns
```
Everything is failing when starting up but in the end it says `successfully initialized` :D 
@0xavi0 can you check where is it trying to create the db file on your env?

**Comment by giubacc** [comment]
Ugh.. I just realized that my build is probably faulty because there is no `WITH_RADOSGW_SFS` flag enabled :) .
So please, ignore all my previous comments until I verified my build configuration.

**Comment by giubacc** [comment]
Completely rebuilt the executable and now it is working as expected.
`rgw_sfs_data_path` is working well when you are not using the default.
LGTM

**Review by giubacc — APPROVED** [review]

**Pull request merged**